test/errchk.go

```diff
@@ -184,47 +203,35 @@ func wantedErrors(file, short string) (errs []wantedError) {
 		if m == nil {
 			continue
 		}
-		all := m[1]
-		mm := errQuotesRx.FindAllStringSubmatch(all, -1)
-		if mm == nil {
-			log.Fatalf("%s:%d: invalid errchk line: %s", file, lineNum, line)
+		rest := m[1]
+		linter := defaultLinter
+		if lm := linterPrefixRx.FindStringSubmatch(rest); lm != nil {
+			linter = lm[1]
+			rest = rest[len(lm[0]):]
+		}
+		rx, err := strconv.Unquote(strings.TrimSpace(rest))
+		if err != nil {
+			log.Fatalf("%s:%d: invalid errchk line: %s, %v", file, lineNum, line, err)
 		}
-		for _, m := range mm {
-			replacedOnce := false
-			rx := lineRx.ReplaceAllStringFunc(m[1], func(m string) string {
-				if replacedOnce {
-					return m
-				}
-				replacedOnce = true
-				n := lineNum
-				if strings.HasPrefix(m, "LINE+") {
```

Comment:
as far as I could tell, nothing was using this logic that appears to have to do with matching line numbers.